bip-0047.mediawiki

```diff
@@ -365,6 +365,37 @@ Bob detects notification transactions by adding his payment code identifier to h
 
 Alice's wallet should spend the notification change output at the next appropriate opportunity.
 
+==Version 3==
+
+A user of version 3 payment codes may use the same payment code to receive payments on different blockchains without address collisions.
+
+Version 3 payment codes behave identifically to version 2 payment codes, except as modified below.
+
+===Representation===
+
+====Binary Serialization====
+
+* Byte 0: version. required value: 0x03
+
+===Protocol===
+
+====Sending====
+
+
+# Each time Alice wants to initiate a transaction to Bob, Alice derives a unique P2PKH address for the transaction using ECDH as follows:
+## Alice selects the 0th private key derived from her payment code: <pre>a</pre>
+## Alice selects the next unused public key derived from Bob's payment code, starting from zero: <pre>B, where B = bG</pre>
+### The "next unused" public key is based on an index specific to the Alice-Bob context, not global to either Alice or Bob
+## Alice selects the 4 byte registered coin type for the transaction based on SLIP-0044: <pre>t</pre>
+## Alice calculates a secret point: <pre>S = aB</pre>
+## Alice calculates a scalar shared secret using the x value of S: <pre>s = SHA256(HMAC-SHA512(Sx, t))</pre>
```

Comment:
What is the goal of using HMAC-SHA512 here? Why was SHA512 chosen as a hash function?

Reply:
Note: Adding HMAC-SHA512 doesn't require the addition of any new functions to BIP47 implementations, as it is used for calculating the blinding factor of the notification transaction. 👍

Reply:
HMAC-SHA512 is also part of BIP-32, so it's a hash function you can guarantee is available in any HD wallet.